source/javascripts/components/StepItem/StepItem.tsx

```diff
@@ -12,60 +12,70 @@ import verifiedIcon from "../../../images/step/badge-verified.svg";
 import deprecatedIcon from "../../../images/step/badge-deprecated.svg";
 
 type StringProps = {
-    alwaysLatest: string
-}
+	alwaysLatest: string;
+};
 
 type StepItemProps = {
-  step: Step,
-  strings: StringProps,
-  selected: boolean,
-  highlightVersionUpdate: boolean,
-  stepIndex: number,
-  onSelected: (step: Step, index: number) => void
-}
+	step: Step;
+	title: string;
+	version: string;
+	strings: StringProps;
+	selected: boolean;
+	highlightVersionUpdate: boolean;
+	stepIndex: number;
+	onSelected: (step: Step, index: number) => void;
+};
 
 export const normalizeIconUrl = (step?: Step): string | undefined => {
-  if (!step) {
-    return;
-  }
+	if (!step) {
+		return;
+	}
 
-  const stepIconURL = step.iconURL();
-  return stepIconURL || defaultStepIcon;
+	const stepIconURL = step.iconURL();
+	return stepIconURL || defaultStepIcon;
 };
 
-const tabIndex = (selected: boolean): number => selected ? -1 : 0;
+const tabIndex = (selected: boolean): number => (selected ? -1 : 0);
 
-const stepVersion = (step: Step, highlightVersionUpdate: boolean) => (highlightVersionUpdate
-  ? <Text>{step.version}</Text>
-  : <Badge backgroundColor="red-3" color="white">
-      <Icon name="ArrowUp" />
-      {step.version}
-    </Badge>
-);
+const stepVersion = (step: Step, highlightVersionUpdate: boolean) =>
+	highlightVersionUpdate ? (
+		<Text>{step.version}</Text>
+	) : (
+		<Badge backgroundColor="red-3" color="white">
+			<Icon name="ArrowUp" />
+			{step.version}
+		</Badge>
+	);
 
 const StepItem: FunctionComponent<StepItemProps> = ({
-  step,
-  strings,
-  selected,
-  highlightVersionUpdate,
-  stepIndex,
-  onSelected
-} : StepItemProps) => (
-  <button className="step" tabIndex={tabIndex(selected)} onClick={() => onSelected(step, stepIndex)}>
-    <img className="icon" src={normalizeIconUrl(step)} />
-    <span className="info">
-      <strong>
-        <Text className="title">{step.displayName()}</Text>
-        {step.isVerified() && <img className="verified" src={verifiedIcon} />}
-        {step.isDeprecated() && <img className="deprecated" src={deprecatedIcon} />}
-      </strong>
-      <em className="version">
-        {step.requestedVersion()
-            ? stepVersion(step, highlightVersionUpdate)
-            : <Text>{strings.alwaysLatest + (step.version && ` (${step.version})`)}</Text>}
-      </em>
-    </span>
-  </button>
+	step,
+	title,
+	version,
+	strings,
+	selected,
+	highlightVersionUpdate,
+	stepIndex,
+	onSelected
+}: StepItemProps) => (
+	<button className="step" tabIndex={tabIndex(selected)} onClick={() => onSelected(step, stepIndex)}>
+		<img className="icon" src={normalizeIconUrl(step)} />
+		<span className="info">
+			<strong>
+				<Text className="title" ellipsis>
+					{title}
+				</Text>
+				{step.isVerified() && <img className="verified" src={verifiedIcon} />}
```

Comment:
don't we need to pass these down as props as well? `isVerified` and `isDeprecated`